### PR TITLE
Fix msg decode error

### PR DESCRIPTION
**Pull request by colinl**
Fixes issue #26

### CODE REVIEW

**Comment by colinl** [comment]
Hi Nicholas, are you able to merge this?

**Comment by colinl** [comment]
Just realised I meant to ask you to check that writes still work ok, I don't have the kit to check that.
Thanks.

**Comment by colinl** [comment]
Ping?

**Comment by njh** [comment]
Hi Colin,

Sorry, for some reason notifications were turned off for this repository and I didn't get your messages.

Thank you very much for this fix. It looks like a good improvement.
Would it be possible to add a test for this?

nick.

File: lib/base/communication.js [hunk 1]
```diff
@@ -28,29 +29,46 @@ var sendCommandToSocket = function (options, socket, callback) {
...
+            debug('Extracted header', header);
+            if (header.ret < 0) {
+                callbackOnce({
+                    msg: 'Communication Error. Received ' + header.ret,
+                    header: header,
+                    options: options
+                });
+            }
+            if (header.payload > 0) {
+                var payload = buffer.slice(24,24+header.payload).toString('utf8');
```
Comment (njh) [inline lib/base/communication.js]: should this make use of ```headerLength```?

Comment (colinl) [inline lib/base/communication.js]: Yes

Comment (colinl) [inline lib/base/communication.js]: I have changed it in my running system, I can't see how it should cause any problems but want it to run overnight to make sure the condition that caused the problem will occur to make sure it is ok.

**Comment by njh** [comment]
Just realised I wrote a similar test for a STOMP server implementation - some problem - checking for a message split over two packets:

https://github.com/bbc/node-radiovis-stomp-server/blob/master/test/stomp/connection.js#L74

If you are not sure about writing tests - I can help. 

Do you have an example data for what a message split over multiple packets looks like?

**Comment by colinl** [comment]
I will have to get back into it to remind myself what was happening. I don't think it was one message split over two packets, but a 'holding' header with no data, followed in the same packet by the real header and payload. The result was that the second header was interpreted as part of the assumed payload for the first header. I have no experience of the test harness (and not much in js).

**Comment by colinl** [comment]
I have confirmed that the problem occurred when a header with an empty message is immediately followed by a good message/header and the combined sequence is received by the socket in one packet. So the failing condition is a received packet starting with a header of the form:
        'header': {
            'version': 0,
            'payload': -1,
            'ret': 0,
            'controlflags': 0,
            'size': 0,
            'offset': 0
        }
followed a by good message and header. Is it simple to include a test of that form?

**Comment by njh** [comment]
Yes, I will try and write some tests for this.

**Comment by colinl** [comment]
Thanks, I will watch what you do and hopefully learn.  Don't worry about the delay I was busy doing other things anyway. Will try and look at the Automatic Retry issue shortly, but it is summer so often better things to do :)

**Pull request merged**